### ISSUE TITLE
connectionClosed for TLS Report

### ISSUE THREAD

**Issue by madic-creates** [hardenize/hardenize-public#54]
I'm not able to get a report for my TLS settings.
It's saying "connection Closed".

In my nginx error log i get the following messages:

2019/03/18 00:08:03 [crit] 21810#21810: *1349 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client:
2600:1f18:753:1b00:eb42:a50a:13c8:2f67, server: [::]:443
2019/03/18 00:08:03 [crit] 21827#21827: *1350 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client:
2600:1f18:753:1b00:eb42:a50a:13c8:2f67, server: [::]:443
2019/03/18 00:08:03 [crit] 21827#21827: *1353 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client: 18.233.176.231, server: 0.0.0.0:443
2019/03/18 00:08:03 [crit] 21777#21777: *1357 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client: 18.233.176.231, server: 0.0.0.0:443
2019/03/18 00:35:50 [crit] 41349#41349: *13120 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client:
2600:1f18:753:1b00:eb42:a50a:13c8:2f67, server: [::]:443
2019/03/18 00:35:50 [crit] 41283#41283: *13121 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client:
2600:1f18:753:1b00:eb42:a50a:13c8:2f67, server: [::]:443
2019/03/18 00:35:50 [crit] 41316#41316: *13122 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client: 18.233.176.231, server: 0.0.0.0:443
2019/03/18 00:35:50 [crit] 41250#41250: *13124 SSL_do_handshake() failed
(SSL: error:0607C084:digital envelope routines:EVP_CIPHER_CTX_ctrl:ctrl
not implemented error:14202006:SSL routines:derive_secret_key_and_iv:EVP
lib) while SSL handshaking, client: 18.233.176.231, server: 0.0.0.0:443

Tested domain: www.geekbundle.org

**Comment by ivanr** [comment]
Fixed in v1.1053.1. Thanks!

**Comment by madic-creates** [comment]
Well..uhm...great. But what was the problem? Something I did wrong on my side and you had to do an exception?

**Comment by ivanr** [comment]
I had to work around the fact that your server closes TCP connection (either always, or in some situations) when it's unable to complete the handshake. Before, when that happened, we would consider it to be a network an give up, which is why you were not seeing the TLS parts in your report. Because we now consider that to be a handshake failure, we are able to show complete results.

Technically speaking, that behaviour breaks the TLS spec, although in a way that the usual communication patterns won't be affected. I've never seen the above error message (and there's not much on the Internet), so I don't know what is the root cause. Recently, I've noticed many servers closing connections instead of refusing handshakes (with TLS alerts), so my guess would be that there was a change in an underlying library, perhaps OpenSSL.

In a nutshell, if the cipher suites we show match what you expect, then there's nothing else to do.
